services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -58,18 +58,23 @@ open class ApplicationService(
     open fun create(application: Application, userId: String): Application {
         logger.info("Creating a new application for user $userId")
 
+        val hankeTunnus =
+            application.hankeTunnus ?: throw ApplicationArgumentException("Missing hankeTunnus")
+
         validateGeometry(application.applicationData) { validationError ->
             "Invalid geometry received when creating a new application for user $userId, reason = ${validationError.reason}, location = ${validationError.location}"
         }
 
         val hanke =
-            hankeRepository.findByHankeTunnus(application.hankeTunnus)
-                ?: throw HankeNotFoundException(application.hankeTunnus)
-
-        application.applicationData.areas?.let { areas ->
-            checkApplicationAreasInsideHankealue(hanke.id!!, areas) { applicationArea ->
-                "Application geometry doesn't match any hankealue when creating a new application for user $userId, " +
-                    "hankeId = ${hanke.id}, application geometry = ${applicationArea.geometry.toJsonString()}"
+            hankeRepository.findByHankeTunnus(hankeTunnus)
+                ?: throw HankeNotFoundException(hankeTunnus)
+
+        if (!hanke.generated) {
```

Comment:
Is creating a new application allowed for an automatically generated hanke?

Reply:
`checkApplicationAreasInsideHankealue` is also called when updating an application and when sending the application to Allu. Both are relevant cases for a johtoselvityshakemus with a generated hanke, so those checks should be disabled for generated hanke as well.

There should be tests for all of these cases.

Reply:
> `checkApplicationAreasInsideHankealue` is also called when updating an application and when sending the application to Allu. Both are relevant cases for a johtoselvityshakemus with a generated hanke, so those checks should be disabled for generated hanke as well.
> 
> There should be tests for all of these cases.

Added the same condition for updating and sending. Includes tests to verify the skip.